client/rest-high-level/src/main/java/org/opensearch/client/indices/PutMappingRequest.java

```diff
@@ -134,26 +161,44 @@ public PutMappingRequest source(String mappingSource, XContentType xContentType)
      */
     public PutMappingRequest source(XContentBuilder builder) {
         this.source = BytesReference.bytes(builder);
-        this.xContentType = builder.contentType();
+        MediaType mediaType = builder.contentType();
+        if (mediaType instanceof XContentType == false) {
```

Comment:
Why do we need this check ?

Reply:
It's not a requirement so much as a "graceful" fail early check since we still so heavily depend on XContent and do not yet support other media types. This way a "nice" exception is thrown if a plugin implements an unsupported media type.

Reply:
> But may at this moment, since MediaType is coupled with XContent, this check may not be necessary?

I agree w/ this. The build will fail in plugins anyway if it's not an XContentType so i'll remove this check for now until we further decouple the XContent dependency